src/test/java/redis/embedded/RealRedisClusterTest.java

```diff
@@ -0,0 +1,157 @@
+        instance = new RealRedisCluster(servers);
+
+        //when
+        instance.create();
+
+        //then
+        for (Redis s : servers) {
+            verify(s).start();
+        }
+    }
+
+    @Test
+    public void stopShouldStopEntireCluster() throws Exception {
+        //given
+        final List<Redis> servers = Arrays.asList(master1, master2, master3);
+        instance = new RealRedisCluster(servers);
+
+        //when
+        instance.stop();
+
+        //then
+        for (Redis s : servers) {
+            verify(s).stop();
+        }
+    }
+
+    @Test
+    public void numberOfNodeShouldBeMoreThanThree() throws Exception {
+        //given
+        final List<Redis> oneServer = Arrays.asList(master1);
+        //when
+        try {
+            instance = new RealRedisCluster(oneServer);
+            fail();
+        } catch (EmbeddedRedisException e) {
+            assertThat(e.getMessage(), equalTo("Redis Cluster requires at least 3 master nodes."));
+        }
+
+        final List<Redis> twoServers = Arrays.asList(master1, master2);
+        try {
+            instance = new RealRedisCluster(twoServers);
+            fail();
+        } catch (EmbeddedRedisException e) {
+            assertThat(e.getMessage(), equalTo("Redis Cluster requires at least 3 master nodes."));
+        }
+    }
+
+    @Test
+    public void numberOfReplicatesShouldBeMoreThatOne() throws Exception {
+        final List<Redis> threeServers = Arrays.asList(master1, master2, master3);
+        try {
+            instance = new RealRedisCluster(threeServers, 0);
+            fail();
+        } catch (EmbeddedRedisException e) {
+            assertThat(e.getMessage(), equalTo("Redis Cluster requires at least 1 replication."));
+        }
+    }
+
+    @Test
+    public void numberOfReplicatesShouldBeLessThanNumberOfServers() throws Exception {
+        final List<Redis> threeServers = Arrays.asList(master1, master2, master3);
+        try {
+            instance = new RealRedisCluster(threeServers, 10);
+            fail();
+        } catch (EmbeddedRedisException e) {
+            assertThat(e.getMessage(), equalTo("Redis Cluster requires number of replications less than number of nodes - 1."));
+        }
+    }
+    @Test
+    public void isActiveShouldCheckEntireClusterIfAllActive() throws Exception {
+        //given
+        given(master1.isActive()).willReturn(true);
+        given(master2.isActive()).willReturn(true);
+        given(master3.isActive()).willReturn(true);
+        final List<Redis> servers = Arrays.asList(master1, master2, master3);
+        instance = new RealRedisCluster(servers);
+
+        //when
+        instance.create();
+        boolean isActive = instance.isActive();
+
+        //then
+        for (Redis s : servers) {
+            verify(s).isActive();
+        }
+        assertThat(isActive, equalTo(true));
+
+    }
+
+    @Test
+    public void createShouldStartEntireCluster() throws Exception {
```

Comment:
Here, I think it makes sense to wrap create() as the first instruction in start(). Then this test should be named "shouldStartRedisCluster".
Additionally, does jedis.hset() enough to test the Cluster has been setup correctly? (Not only contituent parts)